core/examples/driver/main.cpp

```diff
@@ -144,80 +141,43 @@ int main(int /*argc*/, char** /*argv*/)
       }
       else
       {
-        Module* module = ModuleRegistry::GetModule(idOrName);
+        Module* module = frameworkContext->GetModule(idOrName);
         if (!module)
         {
           try
           {
-            std::map<std::string, SharedLibrary>::iterator libIter =
-                libraryHandles.find(sharedLib.GetFilePath(idOrName));
-            if (libIter != libraryHandles.end())
-            {
-              libIter->second.Load();
-            }
-            else
-            {
-              bool libFound = false;
-              for (std::vector<std::string>::const_iterator availableModuleIter = availableModules.begin();
-                   availableModuleIter != availableModules.end(); ++availableModuleIter)
-              {
-                if (*availableModuleIter == idOrName)
-                {
-                  libFound = true;
-                }
-              }
-              if (!libFound)
-              {
-                std::cout << "Error: unknown example module" << std::endl;
-              }
-              else
-              {
-                SharedLibrary libHandle(LIB_PATH, idOrName);
-                libHandle.Load();
-                libraryHandles.insert(std::make_pair(libHandle.GetFilePath(), libHandle));
-              }
-            }
-
-            std::vector<Module*> modules = ModuleRegistry::GetModules();
-            for (std::vector<Module*>::const_iterator moduleIter = modules.begin();
-                 moduleIter != modules.end(); ++moduleIter)
-            {
-              availableModules.erase(std::remove(availableModules.begin(), availableModules.end(), (*moduleIter)->GetName()),
-                                     availableModules.end());
-            }
-
+            /* It is possible to install a bundle based on its name, if its
```

Comment:
This limitation is not described in the comments for InstallBundle. Why does the bundle name need to be the same as the file that contains it (assuming this is what this comment implies)?